### ISSUE TITLE
Flex Attention: Calculates Gradients Even if Input Has requires_grad=False

### ISSUE THREAD

**Issue by cora-codes** [pytorch/pytorch#136101]
### 🐛 Describe the bug

This is a feature request for flex attention to not calculate the gradient of a input if it has `requires_grad=False` I've looking into the kernel and seems like flex attention might actually might make this quite simple compared to other implementations like flash attention, it just might require an additional kernel option.

Let me know you'd like help with this and I can take a stab at it.

It seems like it might be easier to start by not calculating the query gradient and then later add support for calculating key and value gradient separately.

### Versions

2.5.0.dev20240912+cu124

cc @ezyang @chauhang @penguinwu @zou3519 @ydwu4 @bdhirsh @Chillee @drisspg @yanboliang @BoyuanFeng

**Comment by drisspg** [comment]
this is pretty old, and we never ended up implementing its not a bad idea but tbh not sure how common it is to warrant the added complexity
